app/controllers/ems_cloud_controller.rb

```diff
@@ -160,7 +160,7 @@ def ems_cloud_form_fields
                      :provider_id                     => @ems.provider_id ? @ems.provider_id : "",
                      :hostname                        => @ems.hostname,
                      :api_port                        => @ems.port,
-                     :api_version                     => @ems.api_version,
+                     :api_version                     => @ems.api_version ? @ems.api_version : "v2",
```

Comment:
From my testing the angular changes form just according to provider type string, which you change by changing the selectbox of provider type, which is not ideal, cause I wanted to use the 'supports_..' methods we have on models. Maybe it's work in progress?

They don't have the right class since first request doesn't have the provider type set and then the logic is in JS.